app/models/authentication_token.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: authentication_tokens
+#
+#  id           :bigint           not null, primary key
+#  enabled      :boolean          default(TRUE)
+#  expires_at   :date
+#  hashed_token :string
+#  created_at   :datetime         not null
+#  updated_at   :datetime         not null
+#  provider_id  :bigint
+#
+# Indexes
+#
+#  index_authentication_tokens_on_provider_id  (provider_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (provider_id => providers.id)
+#
+
+class AuthenticationToken < ApplicationRecord
+  belongs_to :provider
+
+  before_save :hash_token, if: :hashed_token_changed?
```

Comment:
Typically the caller won't be providing a token, they will want a new random token and I don't think there is a use case for updating a token. Would it be more helpful (and make the interface to this model more explicit) to have a class method that creates a new random token? e.g. https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/models/vendor_api_token.rb#L6-L10